### PR TITLE
Fixed #31088 -- Added support for websearch searching in SearchQuery.

### PR DESCRIPTION
**Pull request by jamesturk**

### CODE REVIEW

**Comment by jamesturk** [comment]
These failures seem to be mysql issues on the test env.  

It doesn't seem like Postgres >=11 is running in the test env, so the tests are skipped.  I have run them locally w/ django-docker-box and they are passing, but I'm not sure what the protocol is here.

File: docs/ref/contrib/postgres/search.txt [hunk 1]
```diff
@@ -80,9 +80,16 @@ looks for matches for all of the resulting terms.
...
+formatted search query with terms and operators.  If ``search_type`` is
+``'websearch'``, then you can provide a formatted search query that resembles
+how you would search most popular search engines.  Read PostgreSQL's
+`Full Text Search docs`_ to learn about differences and syntax. Examples:
 
+.. versionchanged:: 3.1
+
+  Added support for 'websearch' if using Postgres 11 or later.
+
+    Support for spatial functions operating on real geometries was added.
```
Comment (felixxm) [inline docs/ref/contrib/postgres/search.txt]: I'm going to remove this unrelated line.

File: tests/postgres_tests/test_search.py [hunk 2]
```diff
@@ -202,6 +202,28 @@ def test_raw_search_with_config(self):
...
+        searched = line_qs.filter(search=SearchQuery('"body burned" "kneecaps split"', search_type='websearch'))
+        self.assertSequenceEqual(searched, [self.verse1])
+
+    @skipUnlessDBFeature('has_websearch_to_tsquery')
+    def test_web_search_with_config(self):
+        line_qs = Line.objects.annotate(
+            search=SearchVector('scene__setting', 'dialogue', config='french'),
+        )
+        searched = line_qs.filter(
+            search=SearchQuery('cadeau ~beau', search_type='websearch', config='french'),
```
Comment (felixxm) [inline tests/postgres_tests/test_search.py]: `~` is not a supported operator, this should be `-`.

**Comment by felixxm** [comment]
@jamesturk Thanks for this patch :+1: I pushed minor edits to docs and improved tests.

Welcome aboard :boat:

**Comment by gatlanticus** [comment]
I'm not sure if this is the right place to post this, but I am getting an error the following error when trying to use `search_type="websearch"` with PostgreSQL 12:

```
django.db.utils.ProgrammingError: function websearch_to_tsquery(unknown) does not exist
LINE 1: ...or(COALESCE("tree_content"."answer", '')), 'B')), websearch_...
                                                             ^
HINT:  No function matches the given name and argument types. You might need to add explicit type casts.
```

All the other `search_type` s work, so I thought it may be a problem with the patch.
